UI: Ensure Stack is exported from root package exports

aduth opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?

Ensures that the `Stack` component is exported from the `@wordpress/ui` package.

Follow-up to #73308

## Why?

At the moment, the component cannot be used from the package.

## How?

- Adds export in package entrypoint `packages/ui/src/index.ts`
- Adds test coverage to ensure that each component that follows [the documented folder structure](https://github.com/WordPress/gutenberg/blob/trunk/packages/ui/CONTRIBUTING.md#folder-structure) should have its folder referenced in this entrypoint file
- A bit of miscellaneous housekeeping while I was here:
   - Establishes a `CHANGELOG.md`
   - Removes an unreferenced `lock-unlock.ts` file
   - Removes an invalid "Internal dependencies" DocBlock 

## Testing Instructions

- Add an `import { Stack } from '@wordpress/ui';` code snippet in some code file, and a corresponding `"@wordpress/ui": "../ui"` dependencies declaration in that code's closest `package.json`
- Observe that there are no lint errors references an invalid export when running `npm run lint:js`

## Screenshots or screencast <!-- if applicable -->

<img width="566" height="116" alt="image" src="https://github.com/user-attachments/assets/1807ce4c-4258-4b44-8ac8-9aa98a080660" />

aduth committed:
UI: Ensure Stack is exported from root package exports

jsnajdr commented on packages/ui/src/lock-unlock.ts:
```diff
@@ -1,10 +0,0 @@
-	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
-		'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
-		'@wordpress/dataviews'
-	);
```
We can also remove references to `@wordpress/private-apis` from `package.json` and `tsconfig.json` if we don't use it any more.

aduth replied:
Good catch. Fixed in b93be579b94.

mirka commented on packages/ui/src/test/index.test.ts:
```diff
@@ -0,0 +1,20 @@
+	it( 'should export something from each component', async () => {
+		// As described in the CONTRIBUTING.md file, each component should be
+		// exported from an index.ts in its implementation directory.
+		const components = await glob( join( __dirname, '../*/index.ts' ) );
```
I'm paranoid about breaking in Windows again, could this need [this kind of workaround](https://www.npmjs.com/package/fast-glob#how-to-write-patterns-on-windows)? 

For example I'm seeing a normalizer used here:

https://github.com/WordPress/gutenberg/blob/b59a2456328c85e7a3af4440ca8c21eec1209276/packages/wp-build/src/build.mjs#L456-L457

aduth replied:
I feel frustrated at the ambiguity that my answer to your question is "I don't know, and I can't easily confirm" 😕 

For what it's worth, we recently enabled Windows checks on pull requests in #73923, though I need to rebase the branch for that to apply here, and it only covers "Static checks" (including `npm run build`), but not the unit tests, so it wouldn't help here.

aduth replied:
> but not the unit tests, so it wouldn't help here.

Although to that point, it doesn't appear that we run unit tests in a Windows environment [_at any point_](https://github.com/WordPress/gutenberg/blob/c7dff504a2bc4f74de9a242f925e07cd852c5a7f/.github/workflows/unit-test.yml#L35-L45), so we have no guarantees that the tests _currently_ work on Windows machines 😬

aduth replied:
I've been dabbling a bit with using [GitHub Codespaces](https://github.com/features/codespaces) and this seems like it'd be a great use for testing how code behaves on other systems, but sadly it only offers Linux containers at the moment.

aduth replied:
Okay, I understand better now why we have `normalizePath`, and the short answer to your question is "yes".

It's a [requirement of `fast-glob`](https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#pattern-syntax):

>⚠️ Always use forward-slashes in glob expressions (patterns and [ignore](https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#ignore) option). Use backslashes for escaping characters.

I think we won't need these workarounds once we upgrade to a version of Node.js that includes [`fs.glob`](https://nodejs.org/api/fs.html#fsglobpattern-options-callback), which I suspect will handle this better (available in Node.js v22+, see #72973).

aduth replied:
Fixed in 3bba4f579aa

aduth committed:
Remove unused references to private-apis

aduth committed:
Avoid path.join for glob paths

fast-glob requires forward slashes. path.join produces backslashes in Windows environment. Simple fix is to avoid path.join

aduth committed:
Update package-lock.json

mirka commented on packages/ui/src/test/index.test.ts:
```diff
@@ -0,0 +1,20 @@
+	it( 'should export something from each component', async () => {
+		// As described in the CONTRIBUTING.md file, each component should be
+		// exported from an index.ts in its implementation directory.
+		const components = await glob( __dirname + '/../*/index.ts' );
```
Wouldn't this still be mixed slash?

```
C:\foobar\gutenberg\packages\ui\src\test/../*/index.ts
```

Maybe `cwd` in the options? That pattern seems to be used in a few other places.

```suggestion
		const components = await glob( '*/index.ts', { 
			cwd: join( __dirname, '..' ) 
		} );
```

aduth replied:
> Wouldn't this still be mixed slash?

Argh, yeah, you're right.

> Maybe cwd in the options? That pattern seems to be used in a few other places.

This could work 👍

aduth committed:
Use cwd option with fast-glob to avoid mixed slashes

Co-authored-by: Lena Morita <lena@jaguchi.com>

aduth committed:
Fix lint errors
